_docs/rfc/imports.md

```diff
@@ -0,0 +1,963 @@
+info across all the different regions.
+
+**Reusing common variables**
+
+Reusing common variables depends on the scope of `locals`. If the scope of `locals` is shared across the environment,
+then you can define the common variable in `locals` blocks to share across the entier environment. If instead the scope
+of `locals` is per file, we either:
+
+- Define all the code for the single environment in a single file.
+- Implement [globals](#globals) in a way that can be shared across the environment.
+
+**Reusing dependencies**
+
+Reusing dependencies is not a problem in this approach because the namespace for the environment is shared. That is, you
+can reference any of the other `module` blocks to hook up the dependency within a single environment. For example, if
+you had two modules `app` and `mysql` which depend on the `vpc` module, you could define the config as follows:
+
+```
+module "vpc" {
+  # args omitted for brevity
+}
+
+module "app" {
+  # args omitted for brevity
+  vpc_id = module.vpc.outputs.vpc_id
+}
+
+module "mysql" {
+  # args omitted for brevity
+  vpc_id = module.vpc.outputs.vpc_id
+}
+```
+
+This example reuses the outputs of `module.vpc` across the two modules, which is the equivalent of having the `vpc`
+`dependency` block redefined in the two module configs.
+
+**Keeping remote state configuration DRY**
+
+This example is covered in [the original issue that proposed this
+idea](https://github.com/gruntwork-io/terragrunt/issues/759).
+
+
+#### hcldecode or read_terragrunt_config helper function
+
+Instead of defining a dedicated block for `import`, we could define a helper function that parses the relevant config.
+For example, the explicit triple import example in the [hierarchical variables use
+case](#hierarchical-variables-included-across-multiple-terragrunt-hcl-files) can be implemented as:
+
+```
+locals {
+  root_config = read_terragrunt_config("../../../root.hcl")
+  region_config = read_terragrunt_config("../../region.hcl")
+  env_config = read_terragrunt_config("../env.hcl")
+}
+
+inputs = merge(
+  local.root_config.inputs,
+  local.region_config.inputs,
+  local.env_config.inputs,
+  {
+    # args to module
+  },
+)
+```
+
+Pros:
+
+- Relatively simple implementation. It is significantly easier to add helper functions than it is to introduce new
+  blocks in terragrunt.
+- Can address all the same use cases as `import` blocks.
+
+Cons:
+
+- Everything has to be explicit in the config. We can not support automatic merging capabilities when using helper
+  functions, as you can't manipulate the config in a helper. This can lead to very verbose configurations if one wants
+  to use this in place of `include`.
+
+Note that to take full advantage of this approach, all the blocks in `terragrunt.hcl` need to be redefined as attributes
+so that we can use assignment to override them.
+
+Let's walk through a few more of the use cases:
+
+**Keeping remote state configuration DRY**
+
+parent
```

Comment:
There is no more parent or child with the new approach. All the configs for an account are in the same folder and there is one `remote_state` config in one of those `.hcl` files that looks something like this:

```hcl
remote_state {
  backend = "s3"
  config {
    bucket = "foo"
    region = "us-east-1"
    # module_name returns the name of the current module (e.g., frontend-app, mysql, etc)
    key = "${module_name()}/terraform.tfstate"
  }
}
```

The new `module_name()` helper is how we ensure each module ends up in a different state file.

Reply:
This is actually the section for `read_terragrunt_config`, not the approach for single tg file. See line 709

Reply:
Doh!

Reply:
Ah I just realized why you got confused. The bold font is exactly the same as `####`. Updated to use italics for the bold heading I was using previously: d1aa8f0